spark/src/main/java/org/apache/iceberg/spark/source/BatchDataReader.java

```diff
@@ -65,8 +68,10 @@
           // Spark eagerly consumes the batches. So the underlying memory allocated could be reused
           // without worrying about subsequent reads clobbering over each other. This improves
           // read performance as every batch read doesn't have to pay the cost of allocating memory.
-          .reuseContainers()
-          .build();
+          .reuseContainers();
+
+      iter = nameMapping != null ?
+          builder.withNameMapping(NameMappingParser.fromJson(nameMapping)).build() : builder.build();
```

Comment:
I'd prefer to keep the name mapping and building separate. There's no need to mix these together.

```java
if (nameMapping != null) {
  builder.withNameMapping(NameMappingParser.fromJson(nameMapping));
}

iter = builder.build();
```